frontend/src/pages/SignUp.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { TextField, Button, Typography, Container, Box } from '@mui/material';
+import {apiCall} from 'utility/UseApi';
+// Define the config object with field properties
+const fieldConfig = {
+  username: {
+    label: 'Username',
+    name: 'username',
+    type: 'text',
+    required: true,
+  },
+  email: {
+    label: 'Email',
+    name: 'email',
+    type: 'email',
+    required: true,
+  },
+  password: {
+    label: 'Password',
+    name: 'password',
+    type: 'password',
+    required: true,
+  },
+  confirm_password: {
+    label: 'Confirm Password',
+    name: 'confirm_password',
+    type: 'password',
+    required: true,
+  },
+};
+
+function SignUp() {
+  const [formData, setFormData] = useState({
+    username: '',
+    email: '',
+    password: '',
+    confirm_password: ''
+  });
+  const [message, setMessage] = useState('');
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (e) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    // Check if passwords match
+    if (formData.password !== formData.confirm_password) {
```

Comment:
All fields should be validated when the user is entering the values and display the error with the field.

Reply:
If there is any error, we need to disable the submit button.